### PR TITLE
[RFC] Add an action, append-and-select, which appends the current query to the input and selects it

### PR DESCRIPTION
**Pull request by rdeaton**
Demo: https://asciinema.org/a/Te9lQZ5vOUV0pz4KNZlq9b4rd

I built this to use as part of interactive pipelines where I wish to add tags, like a cli bookmarking tool. It's not the cleanest implementation, but I think its a nice feature that I haven't found in any similar tools.

If the feature itself is interesting but you'd like to discuss how to clean the implementation up, I'm happy to do so.

### CODE REVIEW

**Comment by lotabout** [comment]
@rdeaton Looks cool!

About the feature: action `append-and-select` seems to be too specific to me. It is a combination of `append`, `select` and `clear-query`. Since skim now supports action contatenation(e.g. `toggle+up`), I was wondering if it is better to implement the actions separately and use a pipeline 
to achieve `append-and-select`, for example: `append-and-focus+toggle+clear-query`, So that we could handle broader situations.

File: src/model.rs [hunk 1]
```diff
@@ -384,6 +385,33 @@ impl Model {
...
+                            false,
+                            &Vec::new(),
+                            &Vec::new(),
+                            &Regex::new("").unwrap(),
+                            (std::usize::MAX, append_idx),
+                        );
+                        append_idx += 1;
+                        let item = Arc::new(item);
+                        v.push(item.clone());
+                        self.item_pool.append(v);
```
Comment (lotabout) [inline src/model.rs:406]: Currently, In interactive mode, item_pool will be cleared on cmd_query change. But of course that won't be a good use case of `append-and-select` and could be ignored right now.

**Comment by rdeaton** [comment]
I agree with the thought on doing it as a series of pipeline actions, and I actually tried doing something similar initially, but ran into some issues getting it there due to the focus issue. I like the idea of backing this out to just append-and-focus, I'll see what I can do in the next couple of days.

In an ideal situation, I'd be able to use enter as a key that does what is featured in this PR, but also allows you to exit if you hit enter with no query line selected, which was my ultimate goal. I'm not sure there's a way to do that with the pipelined actions, but open to ideas

**Comment by rdeaton** [comment]
Ah I've re-discovered why I abandoned that approach. I can't seem to find a good way to force the focus onto the new item. If I just use `selection.append_sorted_items`, the focus isn't guaranteed to update to what the query has. The matcher would probably pick the right one, but its run asynchronously, and so  the pipeline will finish executing before the matcher returns.

I have added a commit to clear out the line killing, we can easily do that with the pipeline.

**Comment by lotabout** [comment]
@rdeaton 

> In an ideal situation, I'd be able to use enter as a key that does what is featured in this PR, but also allows you to exit if you hit enter with no query line selected, which was my ultimate goal. I'm not sure there's a way to do that with the pipelined actions, but open to ideas

I think we could add an action like `accept-if-no-query` and put it before all the other actions. Or `onEmptyQuery(accept)` kind of thing.

**Comment by lotabout** [comment]
@rdeaton I abandoned the "focus" approach too, sadly.

I was hoping the "focus" approach could serve in broader cases, mainly in single-selection where user might want to append-and-accept. But it turns out the async nature of skim made it so hard to focus on a specific item.

 Anyway,
1. I've submit another PR: https://github.com/lotabout/skim/pull/187 to implement the `append-and-select` action, dealing with some UI related things.
2. This PR: https://github.com/lotabout/skim/pull/184 implement the `if-query-empty` action, so this combination might meet your need: `--bind 'enter:if-query-empty(accept)+append-and-select+unix-line-discard'`

**Pull request closed without merging**